photon-core/src/main/java/org/photonvision/vision/pipe/impl/CropPipe.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) Photon Vision.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package org.photonvision.vision.pipe.impl;
+
+import edu.wpi.first.math.MathUtil;
+import org.opencv.core.Mat;
+import org.opencv.core.Rect;
+import org.photonvision.vision.opencv.CVMat;
+import org.photonvision.vision.pipe.CVPipe;
+
+public class CropPipe extends CVPipe<CVMat, CVMat, Rect> {
+    public CropPipe(int width, int height) {
+        this.params = new Rect(0, 0, width, height);
+    }
+
+    @Override
+    protected CVMat process(CVMat in) {
+        Mat mat = new Mat();
+        in.getMat().copyTo(mat);
+        if (fullyCovers(params, mat)) {
+            return new CVMat(mat);
+        }
+
+        int x = MathUtil.clamp(params.x, 0, mat.width());
+        int y = MathUtil.clamp(params.y, 0, mat.height());
+        int width = MathUtil.clamp(params.width, 0, mat.width() - x);
+        int height = MathUtil.clamp(params.height, 0, mat.height() - y);
+
+        return new CVMat(mat.submat(y, y + height, x, x + width));
```

Comment:
This looks like a waste of memory. By copying, then taking a submat, you leave the copy to the GC for cleanup. Why not submat first, then copy? That way, the CVMat can actually free the memory associated with the copied Mat.

Reply:
Why are we even copying in the first place?

Reply:
> Why are we even copying in the first place?

I did that because we needed to release the frame later

Reply:
Don't copy at all here. That will fix the memory leaks I suspect.